tests/views/test_general_user_data_view.py

```diff
@@ -185,3 +193,27 @@ def get_course_rec(course_id, course_list):
             for course_enrollment in CourseEnrollment.objects.filter(user=user_model):
                 # Test that the course id exists in the data
                 assert get_course_rec(course_enrollment.course_id, rec['courses'])
+
+    @pytest.mark.parametrize('search_term', SEARCH_TERMS)
+    def test_get_search(self, search_term):
+        """
+        Based on a SEARCH_TERMS data set, we query the endpoint with search
+        terms and we compare with the expected results.
+        """
+        request_path = self.request_path + '?search=' + search_term['term']
+        request = APIRequestFactory().get(request_path)
+        force_authenticate(request, user=self.staff_user)
+        view = self.view_class.as_view({'get': 'list'})
+        response = view(request)
+        assert response.status_code == 200
+        if not is_multisite():
```

Comment:
This is a cosmetic change that can be addressed on a future PR. But generally an `if` statement, means that there are two separate test cases. 

```python
    @pytest.mark.skipif(organizations_support_sites(),
                        reason='Organizations support sites')
    @pytest.mark.parametrize('search_term', SEARCH_TERMS)
    def test_get_search(self, search_term):
        """
        Based on a SEARCH_TERMS data set, we query the endpoint with search
        terms and we compare with the expected results.
        """
        request_path = self.request_path + '?search=' + search_term['term']
        request = APIRequestFactory().get(request_path)
        force_authenticate(request, user=self.staff_user)
        view = self.view_class.as_view({'get': 'list'})
        response = view(request)
        assert response.status_code == 200
        assert response.data['count'] == search_term['expected_result']
        assert len(response.data['results']) == search_term['expected_result']

    @pytest.mark.skipif(not organizations_support_sites(),
                        reason='Organizations does not support sites')
    @pytest.mark.parametrize('search_term', SEARCH_TERMS)
    def test_get_search(self, search_term):
        """
        Based on a SEARCH_TERMS data set, we query the endpoint with search
        terms and we compare with the expected results.
        """
        request_path = self.request_path + '?search=' + search_term['term']
        request = APIRequestFactory().get(request_path)
        force_authenticate(request, user=self.staff_user)
        view = self.view_class.as_view({'get': 'list'})
        response = view(request)
        assert response.status_code == 200
        # the defaul site is example.com so it won't be find users, and
        # that the expected outcome, we should test with different
        # sites expecting different results, but is out of scope for
        # now.
        assert response.data['count'] == 0
        assert len(response.data['results']) == 0
```

Reply:
👍 Ok, this is good to know, but please, let's address it in a future PR.

Reply:
no problem maxi!

Reply:
This is something that we need to figures out, how we want to run tests that are single site versus multisite. I've done both checking for multisite in order to wire up `UserOrganizationMapping` and constructing `OrganizationCourse` with and without the `sites=[some_site]` parameter. Something that would be a good investigative and decision topic for the unification effort